### PR TITLE
TST: add RC tests for pysat, pysatNASA

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Adds a workflow_dispatch job to test for latest RC versions of pysat and pysatNASA.  Jobs are manually triggered for use in testing RCs of other packages.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

A commit was set to 'push' to test the new workflow.  Logs are at https://github.com/pysat/pysatModels/actions/runs/3362763795

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: .github/workflows/pysat_rc.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# This workflow will install Python dependencies, run tests and lint with a
+# variety of Python versions. For more information see:
```
Comment (jklenzing) [inline .github/workflows/pysat_rc.yml]:
```suggestion
# This workflow will install Python dependencies (including the latest RC 
# for pysat packages) and run tests for supported operating systems.
# For more information see:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 * Maintenance
   * Switched pydineof_dineof load code to use newer `pysat.utils.io.load_netcdf`
   * Updated NEP29 and GitHub actions versions
+  * Added manual test for pysat and pysatNASA Release Candidates
```
Comment (aburrell) [inline CHANGELOG.md:12]: Because this is manual, some instructions for when/how it should be run would be useful.  Perhaps in the Contributing file?

Comment (jklenzing) [reply CHANGELOG.md:12]: This is a good idea.  It should be manually run whenever an RC is up for pysat or pysatNASA.  We may want to add a page there to point to this as part of the wiki.

Comment (jklenzing) [reply CHANGELOG.md:12]: Updated the pysat wiki here: https://github.com/pysat/pysat/wiki/Checklist-for-Release

An example of what the run workflow button looks like is here: https://github.com/pysat/pysat/actions/workflows/stats.yml

Comment (aburrell) [reply CHANGELOG.md:12]: I updated your wiki instructions to make them easier to read and find.

**Review by aburrell — APPROVED** [review]

**Pull request merged**